test/pjrt/test_torchrun.py

```diff
@@ -28,6 +28,41 @@ def test_all_gather(self):
 
     expected = torch.arange(0, expected_world_size, step=1, dtype=torch.float32)
     torch.testing.assert_close(result.cpu(), expected)
+    dist.destroy_process_group()
+
+  def test_all_reduce(self):
+    # The test is inspired by https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_reduce
+    dist.init_process_group('xla', init_method='xla://')
+
+    dist_world_size = xu.getenv_as('WORLD_SIZE', int)
+    devices_per_thread = xr.addressable_device_count()
+    world_size = dist_world_size * devices_per_thread
+    tensors = [torch.arange(2, dtype=torch.int64) + 1 + 2 * r for r in range(world_size)]
```

Comment:
Could you add a comment on what value the tensor should have? It's a bit hard to follow at the moment.

Reply:
done